### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by Eduard-Kozachenko**

### CODE REVIEW

File: app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public void onDestroy() {
+        super.onDestroy();
+        releasePresenter();
+    }
+
+    public void showInformationDialog(String information) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        builder.setMessage(information)
+                .setCancelable(false)
+                .setNeutralButton("OK", (dialog, which) -> dialog.dismiss());
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java]: "OK" it's positive button, not neutral

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+//    @Inject
+//    DialogInfoFragment alertDialogInfo;
+//    @Inject
+//    CollectionPresenter collectionPresenter;
+
+    private TextView tvInfo;
+    private Button btnGenerate;
+
+    private CollectionContract.Presenter mPresenter;
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: Whats it? Remove? it should not be here, Also why you are using "m" in the name of this variable and not using it for other variables?

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+//    }
+
+    @Override
+    public void updateUserInfo(String info) {
+        tvInfo.setText(info);
+    }
+
+    @Override
+    public void showProgressBar() {
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: empty line, use code formatting

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        if (!loadingFragment.isVisible()) {
+            loadingFragment.show(getActivity().getSupportFragmentManager(), "LOADING");
+        }
+    }
+
+    @Override
+    public void hideProgressBar() {
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: empty line

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.eduard.caloriecounter.presentation.model;
+
+public class User {
+
+    public String etWeight,etHeight,etAge;
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java]: Incorrect variable names, "et" means EditText, but this variable is String.

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+     interface View extends BasePresenter.View {
+
+        void updateUserInfo(String info);
+
+    }
+
+    interface Presenter{
+
+//        void onClick(android.view.View view);
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java]: Remove it

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    public void appendValue(String value) {
+
+    }
+
+    @Override
+    public void performCalculation() {
+        String result = "";
+
+        result = mCalculator.operation(mWeight,mHeight,mAge);
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java]: Too much spaces, why you are not formatting your code?

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        android:layout_height="wrap_content"
+        android:gravity="left"
+        android:orientation="horizontal"
+        android:layout_margin="@dimen/margin_ten">
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Choose your gender: "
+            android:paddingRight="20dp"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/collection_of_info_fragment.xml]: use dimens

File: app/src/main/res/layout/alert_info_dialog.xml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        android:src="@drawable/ic_assignment_returned_black_24dp"
+        android:scaleType="fitCenter" />
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="vazno ash ne mogu"
+        android:id="@+id/text_dialog"
+        android:layout_below="@+id/a"
+        android:layout_marginTop="@dimen/margin_twenty"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/alert_info_dialog.xml]: you may use numbers, for example "margin_20"

File: app/src/main/res/layout/alert_info_dialog.xml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        android:background="@color/colorGreen"
+        android:src="@drawable/ic_assignment_returned_black_24dp"
+        android:scaleType="fitCenter" />
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="vazno ash ne mogu"
+        android:id="@+id/text_dialog"
+        android:layout_below="@+id/a"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/alert_info_dialog.xml]: Whats it?

File: app/src/main/res/layout/alert_info_dialog.xml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        android:layout_marginRight="@dimen/margin_four"
+        android:layout_marginBottom="@dimen/margin_twenty"
+        android:textSize="@dimen/text_eighteen"
+        android:textColor="@color/black"
+        android:layout_centerHorizontal="true"
+        android:gravity="center_horizontal" />
+
+    <Button
+        android:layout_width="wrap_content"
+        android:layout_height="30dp"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/alert_info_dialog.xml]: dimens

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+            android:textStyle="bold"
+            ></TextView>
+
+        <RadioButton
+            android:id="@+id/rb_Male"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginRight="16dp"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/collection_of_info_fragment.xml]: dimens

**Review by vladislav-v-rusakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java (hunk 2 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java (hunk 5 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java (hunk 7 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java]: ready

File: app/src/main/res/layout/collection_of_info_fragment.xml (hunk 8 above)
Comment (Eduard-Kozachenko) [reply app/src/main/res/layout/collection_of_info_fragment.xml]: ready

File: app/src/main/res/layout/alert_info_dialog.xml (hunk 9 above)
Comment (Eduard-Kozachenko) [reply app/src/main/res/layout/alert_info_dialog.xml]: ready

File: app/src/main/res/layout/alert_info_dialog.xml (hunk 10 above)
Comment (Eduard-Kozachenko) [reply app/src/main/res/layout/alert_info_dialog.xml]: old version of the alert dialog that I already deleted

File: app/src/main/res/layout/alert_info_dialog.xml (hunk 11 above)
Comment (Eduard-Kozachenko) [reply app/src/main/res/layout/alert_info_dialog.xml]: ready

File: app/src/main/res/layout/collection_of_info_fragment.xml (hunk 12 above)
Comment (Eduard-Kozachenko) [reply app/src/main/res/layout/collection_of_info_fragment.xml]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java (hunk 1 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+//                (dialog, id) -> dialog.cancel());
+//
+//        builder.setNegativeButton(
+//                "Cancel",
+//                (dialog, id) -> dialog.cancel());
+//
+//        AlertDialog alert = builder.create();
+//        alert.show();
+
+        builder.setTitle("Confirm Delete...");
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/base/BaseFragment.java]: Bad idea. Dialog Title and Message should be passed as parameters.

File: app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        presenter = new CollectionPresenter();
+        return inflater.inflate(R.layout.collection_of_info_fragment, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/fragments/CollectionFragment.java]: empty line

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
+package com.eduard.caloriecounter.presentation.model;
+
+public class User {
+
+    public String UserWeight,UserHeight,UserAge;
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java]: Variable name starts with upper case. Fix it.

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
+package com.eduard.caloriecounter.presentation.model;
+
+public class User {
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java:3]: weight, height and age. Where is gender? I think its important.

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+import com.eduard.caloriecounter.presentation.base.BasePresenter;
+import com.eduard.caloriecounter.presentation.utils.CalculatorUserModel;
+
+public class CollectionPresenter extends BasePresenter<CollectionContract.View> implements CollectionContract.Presenter {
+
+    private CollectionContract.Model model;
+
+    public CollectionPresenter() {
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionPresenter.java]: remove empty line

File: app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+package com.eduard.caloriecounter.presentation.utils;
+import com.eduard.caloriecounter.presentation.model.User;
+import com.eduard.caloriecounter.presentation.presenter.CollectionContract;
+
+public class CalculatorUserModel implements CollectionContract.Model{
+
+    User strWeight;
+    User strHeight;
+    User strAge;
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java]: Strange. You may use only one variable here: `User user = new User()`.

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+    public String UserWeight,UserHeight,UserAge;
+    public static final int MAX_DECIMAL_DIGITS = 1;
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java:6]: Also you may use default constants for age, weight, height and etc. This way app may perform calculation without waiting for user input.

File: app/src/main/res/layout/collection_of_info_fragment.xml [hunk 20]
```diff
@@ -0,0 +1,138 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/layout/collection_of_info_fragment.xml:2]: This Layout is redundant. You may use ScrollView as root.

File: app/src/main/res/values/dimens.xml [hunk 21]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <dimen name="margin_4">4dp</dimen>
```
Comment (vladislav-v-rusakov) [inline app/src/main/res/values/dimens.xml:4]: apply code formatting

File: app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java (hunk 15 above)
Comment (vladislav-v-rusakov) [reply app/src/main/java/com/eduard/caloriecounter/presentation/model/User.java]: also apply code formatting. (space after comma)

**Comment by vladislav-v-rusakov** [comment]
@pipi3gunD Done with code review, you may proceed with fixes.

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java [hunk 22]
```diff
@@ -5,20 +5,16 @@
...
          void initView();
          void setViewData(String data);
-
     }
 
     interface Presenter {
-
-        void onClick(android.view.View view);
-
+         void onClick(android.view.View view);
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java]: Non informative method name. AND VERY BAD IDEA TO USE onClick AS NAME OF METHOD!!!!

File: app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java [hunk 23]
```diff
@@ -5,20 +5,16 @@
...
+        void collectionInfo(double weight, double height, double age);
     }
 
     interface Model {
-        String operation();
+        String operation(double weight, double height, double age);
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/presenter/CollectionContract.java]: Non informative method name. What is the operation??

File: app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java [hunk 24]
```diff
@@ -20,46 +20,10 @@
...
-
-        if (fractionals.length() > 0) {
-            // Result has fractionals different than zero - return them!
-            return decimals + "." + fractionals;
-        } else {
-            return decimals;
-        }
+    public String operation(double weight, double height, double age) {
+        User user = new User(weight,height,age);
+        double result =(88.36 + (13.4 * user.getWeight()) + (4.8 * user.getHeight())-(5.7 * user.getAge()));
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java:11]: Hardcoded values. Use constants.

File: app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java [hunk 25]
```diff
@@ -21,9 +21,16 @@
...
+    public String operationMale(double weight, double height, double age) {
         User user = new User(weight,height,age);
         double result =(88.36 + (13.4 * user.getWeight()) + (4.8 * user.getHeight())-(5.7 * user.getAge()));
         return String.valueOf(result);
     }
+
+    @Override
+    public String operationFemale(double weight, double height, double age) {
+        User user = new User(weight,height,age);
+        double result =(447.6 + (9.2 * user.getWeight()) + (3.1 * user.getHeight())-(4.3 * user.getAge()));
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/utils/CalculatorUserModel.java:46]: Hardcoded data. Use Constants.

**Review by vladislav-v-rusakov — APPROVED** [review]

**Pull request merged**